tests/terraform/testutils.go

```diff
@@ -0,0 +1,272 @@
+	if err != nil {
+		return "", "", err
+	}
+	return ip, port, err
+}
+
+func FetchNodeExternalIP(kubeconfig string) []string {
+	cmd := "kubectl get node --output=jsonpath='{range .items[*]} { .status.addresses[?(@.type==\"ExternalIP\")].address}' --kubeconfig=" + kubeconfig
+	time.Sleep(10 * time.Second)
+	res, _ := RunCommand(cmd)
+	nodeExternalIP := strings.Trim(res, " ")
+	nodeExternalIPs := strings.Split(nodeExternalIP, " ")
+	return nodeExternalIPs
+}
+
+func FetchIngressIP(namespace string, kubeconfig string) ([]string, error) {
+	cmd := "kubectl get ingress -n " + namespace + " -o jsonpath='{.items[0].status.loadBalancer.ingress[*].ip}' --kubeconfig=" + kubeconfig
+	res, err := RunCommand(cmd)
+	if err != nil {
+		return nil, err
+	}
+	ingressIP := strings.Trim(res, " ")
+	if ingressIP != "" {
+		ingressIPs := strings.Split(ingressIP, " ")
+		return ingressIPs, nil
+	} else {
+		return nil, nil
+	}
+}
+
+func parseNodes(kubeConfig string, print bool, cmd string) ([]Node, error) {
+	nodes := make([]Node, 0, 10)
+	nodeList := ""
+	res, err := RunCommand(cmd)
+	if err != nil {
+		return nil, err
+	}
+	nodeList = strings.TrimSpace(res)
+	split := strings.Split(nodeList, "\n")
+	for _, rec := range split {
+		if strings.TrimSpace(rec) != "" {
+			fields := strings.Fields(rec)
+			node := Node{
+				Name:       fields[0],
+				Status:     fields[1],
+				Roles:      fields[2],
+				Version:    fields[4],
+				InternalIP: fields[5],
+				ExternalIP: fields[6],
+			}
+			nodes = append(nodes, node)
+		}
+	}
+	if print {
+		fmt.Println(nodeList)
+	}
+	return nodes, nil
+}
+
+func GetNodes(kubeConfig string, print bool) ([]Node, error) {
+	cmd := "kubectl get nodes --no-headers -o wide --kubeconfig=" + kubeConfig
+	return parseNodes(kubeConfig, print, cmd)
+}
+
+func GetWorkerNodes(kubeConfig string, print bool) ([]Node, error) {
+	cmd := "kubectl get node -o jsonpath='{range .items[*]}{@.metadata.name} {@.status.conditions[-1].type} <not retrieved> <not retrieved> {@.status.nodeInfo.kubeletVersion} {@.status.addresses[?(@.type==\"InternalIP\")].address} {@.status.addresses[?(@.type==\"ExternalIP\")].address} {@.spec.taints[*].effect}{\"\\n\"}{end}' --kubeconfig=" + kubeConfig + " | grep -v NoSchedule | grep -v NoExecute"
+	return parseNodes(kubeConfig, print, cmd)
+}
+
+func parsePods(kubeconfig string, print bool, cmd string) ([]Pod, error) {
+	pods := make([]Pod, 0, 10)
+	podList := ""
+	res, _ := RunCommand(cmd)
+	res = strings.TrimSpace(res)
+	podList = res
+
+	split := strings.Split(res, "\n")
+	for _, rec := range split {
+		fields := strings.Fields(string(rec))
+		pod := Pod{
+			NameSpace: fields[0],
+			Name:      fields[1],
+			Ready:     fields[2],
+			Status:    fields[3],
+			Restarts:  fields[4],
+			NodeIP:    fields[6],
+			Node:      fields[7],
+		}
+		pods = append(pods, pod)
+	}
+	if print {
+		fmt.Println(podList)
+	}
+	return pods, nil
+}
+
+func GetPods(kubeconfig string, print bool) ([]Pod, error) {
```

Comment:
Using "get" in Go code isn't idiomtic and a left over from other languages. The more common and idiomatic way is to name the function Pods as that's what is returns. Same for all of the other "get" functions.